java/arcs/android/storage/service/MuxedStorageChannelImpl.kt

```diff
@@ -87,12 +86,12 @@ class MuxedStorageChannelImpl(
       directStoreMuxer: UntypedDirectStoreMuxer,
       scope: CoroutineScope,
       statisticsSink: BindingContextStatisticsSink,
-      callback: IStorageChannelCallback
+      messageCallback: IMessageCallback
     ): MuxedStorageChannelImpl {
-      return MuxedStorageChannelImpl(directStoreMuxer, scope, statisticsSink, callback)
+      return MuxedStorageChannelImpl(directStoreMuxer, scope, statisticsSink)
         .also {
           it.listenerToken = directStoreMuxer.on { message ->
-            callback.onMessage(message.toProto().toByteArray())
+            messageCallback.onMessage(message.toProto().toByteArray())
```

Comment:
I noticed a bug here the other day, but forgot to actually fix it. 

At the moment this is a MuxedProxyMessageProto (serialised to bytes), but actually should be a StorageServiceMessageProto _containing_ a MuxedProxyMessageProto. Would you mind fixing that while you're here (or in a follow up CL)?

Reply:
PTAL, this is my first time dealing with protos